src/plugins/auto/plugin.cpp

```diff
@@ -192,25 +192,32 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
         for (auto&& deviceNameWithID : sameTypeDevices) {
             DeviceIDParser newParsed{deviceNameWithID};
             std::string defaultDeviceID = "";
+            std::string tempDeviceID = "";
             if (newParsed.getDeviceID().empty()) {
                 defaultDeviceID = getDefaultDeviceID(deviceNameWithID);
+                tempDeviceID = defaultDeviceID;
             } else {
-                defaultDeviceID = newParsed.getDeviceID();
+                tempDeviceID = newParsed.getDeviceID();
             }
 
             std::string fullDeviceName = "";
             std::string uniqueName = "";
             if (newParsed.getDeviceName() == "GPU") {
-                auto supportedMetrics = GetCore()->GetMetric(deviceNameWithID, METRIC_KEY(SUPPORTED_METRICS)).as<std::vector<std::string>>();
-                if (std::find(supportedMetrics.begin(), supportedMetrics.end(), METRIC_KEY(FULL_DEVICE_NAME)) != supportedMetrics.end()) {
-                    fullDeviceName = GetCore()->GetMetric(deviceNameWithID, METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>();
+                try {
+                    auto supportedMetrics = GetCore()->GetMetric(deviceNameWithID, METRIC_KEY(SUPPORTED_METRICS)).as<std::vector<std::string>>();
+                    if (std::find(supportedMetrics.begin(), supportedMetrics.end(), METRIC_KEY(FULL_DEVICE_NAME)) != supportedMetrics.end()) {
+                        fullDeviceName = GetCore()->GetMetric(deviceNameWithID, METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>();
+                    }
+                } catch (...) {
+                    LOG_DEBUG_TAG("device %s is not valid", deviceNameWithID.c_str());
+                    continue;
```

Comment:
is it because we suppose that some devices don't support `SUPPORTED_METRICS`?

Reply:
-d MULTI:GPU.0,GPU.1 on a platform actually without GPU.1, without above change, multi load network will throw execption

Reply:
revert the change, -d MULTI:GPU.0,GPU.1 on a platform actually without GPU.1 will throw an exception as before